mathesar_ui/src/stores/table-data/constraints.ts

```diff
@@ -95,6 +120,92 @@ export class ConstraintsDataStore implements Writable<ConstraintsData> {
     return null;
   }
 
+  async add(constraintDetails: Partial<Constraint>): Promise<Partial<Constraint>> {
+    const constraint = await this.api.add(constraintDetails);
+    await this.fetch();
+    return constraint;
+  }
+
+  async remove(constraintId: number): Promise<void> {
+    await this.api.remove(constraintId);
+    await this.fetch();
+  }
+
+  /**
+   * A constraint only matches if the set of its columns strictly equals the set
+   * of columns supplied here. For example, if a constraint is set on three
+   * columns, two of which are passed to this function, that constraint will
+   * _not_ be returned.
+   */
+  constraintsThatMatchSetOfColumns(columns: Column[]): Readable<Constraint[]> {
+    const columnsNames = columns.map((c) => c.name);
```

Comment:
We can optimize this block further by using a Set for `columnsNames`, or even the argument `columns`.